async.el

```diff
@@ -171,7 +171,10 @@ It is intended to be used as follows:
                     (prog1
                         (funcall async-callback proc)
                       (unless async-debug
-                        (kill-buffer (current-buffer))))
+                        ;; we need to check this because theoretically
+                        ;; `async-callback' could've killed it already
+                        (when (buffer-live-p (process-buffer proc))
+                          (kill-buffer (process-buffer proc)))))
```

Comment:
It's a safe no-op to pass killed buffers to `kill-buffer`.

OTOH if `process-buffer` returns `nil`, then `kill-buffer` would try to kill the current buffer instead, and that would be a problem.

So perhaps the comment could be clarified? Thanks.

Reply:
Yea, previously it was just using `(current-buffer)` which could also point really anywhere if the callback changes buffers for example.

So you're saying that the check is useless entirely?  Or we only need to check if `(process-buffer proc)` itself is not nil?  If it's a buffer, killed or not, kill it, but if no buffer is associated, don't call kill-buffer?.  IDK.  The code as is now kind of makes it clear what's the logic.  

How would you improve the comment, or remove it entirely?

Reply:
> How would you improve the comment, or remove it entirely?

The current comment (and the title of this PR) implies that a check is needed before killing a killed buffer, when what the check is actually doing is checking that `proc` still has a buffer at all.

But that's a pedantic reading, so perhaps the current wording is clear enough.